### PR TITLE
Redirect loop when site is offline and home page sends a 404 (solves #7670)

### PR DESCRIPTION
**Pull request by infograf768**
See test instructions in https://github.com/joomla/joomla-cms/issues/7670.

After patching, the site will exit with a message.
"This site is offline and the Home Page sends a 404 error. Please contact a site administrator."

---
#### Steps to reproduce the issue

Create a brand new 3.4.3 site 

> Trash the article related to the default menu item 
> Now set the site to offline from Global Configuration. 
> grab popcorn and watch the front end go to redirect loop. 
#### Expected result

> Site should show offline page
#### Actual result

> Site shows redirect loop. 
#### System information (as much as possible)

Fresh Joomla 3.4.3 install

### CODE REVIEW

**Comment by zero-24** [comment]
hmm works as described.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7676">issues.joomla.org/joomla-cms/7676</a>.</sub>

**Comment by zero-24** [comment]
just added the testinstructions maybe @dhaval-desai can test here also for RTC. Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7676">issues.joomla.org/joomla-cms/7676</a>.</sub>

**Comment by brianteeman** [comment]
Please do not merge - the language string is not in english

On 10 August 2015 at 19:34, zero-24 <notifications@github.com> wrote:

> just added the testinstructions maybe @dhaval-desai
> <https://github.com/dhaval-desai> can test here also for RTC. Thanks.
> ------------------------------
> This comment was created with the J!Tracker Application
> <https://github.com/joomla/jissues> at issues.joomla.org/joomla-cms/7676
> <http://issues.joomla.org/tracker/joomla-cms/7676>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/7676#issuecomment-129559087>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

File: language/en-GB/en-GB.ini [hunk 1]
```diff
@@ -205,6 +205,7 @@ JGLOBAL_LT="&lt;"
 JGLOBAL_NEWITEMSLAST_DESC="New items default to the last position. Ordering can be changed after this item has been saved."
 JGLOBAL_NO_MATCHING_RESULTS="No Matching Results"
 JGLOBAL_NUM="#"
+JGLOBAL_OFFLINE_ERROR="This site is offline and the Home Page sends a 404 error. Please contact a site administrator."
```
Comment (brianteeman) [inline language/en-GB/en-GB.ini]: It should be enough to just display
"This site is offline please contact a site administrator."

**Comment by watchfulli-dev** [comment]
@test Indeed with J3.4.3 trashing article used by default menu item cause a redirect loop. After applying the patch it shows this message "This site is offline and the Home Page sends a 404 error. Please contact a site administrator." no more redirect loop.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7676">issues.joomla.org/joomla-cms/7676</a>.</sub>

**Comment by infograf768** [comment]
>It should be enough to just display
"This site is offline please contact a site administrator."

I disagree. The more important information (as it is in fact aimed at the site administrator), is that the home page sends a 404, as otherwise the message does not help.
If you consider it is not in good English, please propose something which includes the fact that we get a 404 for the Home page.

**Comment by brianteeman** [comment]
The site is offline - it does NOT matter that you tell someone its a 404.
it is enough to say the site is offline.
What is the benefit for saying it is a 404

On 11 August 2015 at 08:45, infograf768 <notifications@github.com> wrote:

> It should be enough to just display
> "This site is offline please contact a site administrator."
>
> I disagree. The more important information (as it is in fact aimed at the
> site administrator), is that the home page sends a 404, as otherwise the
> message does not help.
> If you consider it is not in good English, please propose something which
> includes the fact that we get a 404 for the Home page.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/7676#issuecomment-129727809>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by mbabker** [comment]
IMO if the site is offline, for the general viewing public (unauthenticated users), that should take precedence over the 404 response.  Once a user has logged into the frontend of that offline site, then they should be made aware of the 404 (and I think that'll be handled with normal error handling routines in that case).

**Comment by Bakual** [comment]
> IMO if the site is offline, for the general viewing public (unauthenticated users), that should take precedence over the 404 response.

That's what I wonder with this issue. Why does it even check if the URL is a valid request when the site is offline? Shouldn't it just show that offline page right away when not authenticated? Imho it shouldn't even go into the component dispatching/routing stage and bail out earlier.

**Comment by mbabker** [comment]
> Imho it shouldn't even go into the component dispatching/routing stage and bail out earlier.

You must not be aware of how "bad" Joomla's offline mode is then :wink: 

http://magazine.joomla.org/issues/issue-feb-2011/item/378-how-offline-is-joomla-offline-mode

**Comment by Bakual** [comment]
> You must not be aware of how "bad" Joomla's offline mode is then :wink:

No, I wasn't. Funny thing that :smile:

**Comment by infograf768** [comment]
Folks, at reading http://magazine.joomla.org/issues/issue-feb-2011/item/378-how-offline-is-joomla-offline-mode it looks like this needs another solution than this bandaid.
Closing.

**Comment by infograf768** [comment]
@nikosdion 
Time to implement a real solution I guess. :smiley:

**Comment by nikosdion** [comment]
If you don't care about backwards compatibility and you are happy with me disemboweling the JApplicationCms routing AND adding a hardcoded dependency to com_users deep in its new guts, sure enough, I can make a *real* offline mode.

BUT! *Things Shall Break*.

If your site depends on some plugin or component doing some important work (e.g. unpublish expiring subscriptions, update search indices, backup your site...) on a schedule and the site is REALLY off-line then you will perceive your site as broken. Is this something we are even allowed to think about in the 3.x release cycle?

**Comment by infograf768** [comment]
Sure not, just teasing you... :smiley:

**Comment by nikosdion** [comment]
So let's leave it broken :p

**Comment by infograf768** [comment]
:clap:

**Pull request closed without merging**